test/modules/MetricsValidator/src/TestBase.cs

```diff
@@ -20,14 +22,42 @@ public abstract class TestBase
         protected IMetricsScraper scraper;
         protected ModuleClient moduleClient;
 
+        protected string TestName
+        {
+            get { return this.GetType().ToString().Split('.').LastOrDefault(); }
```

Comment:
Won't `LastOrDefault` return `null` in the default case? And is that what you want?

Reply:
Yea, but I don't think it's possible for split to return an empty array, so it shouldn't ever happen. I could have used First(), but habit.